makefiles/pseudomodules.inc.mk

```diff
@@ -95,6 +95,8 @@ PSEUDOMODULES += picolibc
 PSEUDOMODULES += picolibc_stdout_buffered
 PSEUDOMODULES += pktqueue
 PSEUDOMODULES += posix_headers
+PSEUDOMODULES += posix_sleep
+PSEUDOMODULES += posix_usleep
```

Comment:
Why two modules? Does having both functions in one module really generate such a huge overhead?

Reply:
See third point of PR changes in OP: `The split in 2 separate modules allows to use the rtt backend (and thus low-power) with the sleep function.`

Reply:
So you can now only use either `posix_sleep` or `posix_usleep`?

Reply:
> So you can now only use either posix_sleep or posix_usleep?

Yes. What do you mean with this question ? You want 2 separate submodules and no pseudo-modules ?

Reply:
I was wondering, 

1. If having two modules is sensible with regards of POSIX usability (we don't have a subdivision in multiple modules for other POSIX functions either) and
2. With this new information you gave me, if this is POSIX compliant and practical. What if a user needs both functions when porting a library? Can't there be at least an adaption layer to use `posix_usleep()` for `posix_sleep()` when both `posix_sleep` and `posix_usleep` are compiled in so you still can use both?

Reply:
> What if a user needs both functions when porting a library?

```
USEMODULE += posix_sleep
USEMODULE += posix_usleep
```
also works. I need to check the effects on the underlying timers (`ZTIMER_USEC` and `ZTIMER_MSEC`). While usec will use a high-speed timer backend without low-power capabilities, I'm unsure if msec will still use the rtt backend. Maybe yes, and in this case there would be no need for the split `posix_sleep`/`posix_sleep`, which would simplify this PR. @kaspar030 what do you think ?

Reply:
Now I'm confused :sweat_smile:. Are those modules mutually exclusive or not?

Reply:
> Are those modules mutually exclusive or not?

After discussing with @fjmolinas IRL, this is not the case. So my initial assumption, which lead to the split in 2 modules was wrong. I'll rework that PR.